### DISCUSSION TITLE
Flatpak packaging PoC

### DISCUSSION THREAD

**Discussion by tinywrkb** [any1/wlvncc#2] [Show and tell]
This is not a feature request, I'm just sharing [my Flatpak manifest for wlvncc](https://github.com/tinywrkb/flatpaks/blob/master/com.github.any1.wlvncc/com.github.any1.wlvncc.yaml).  
Being able to decouple the app from the system is pretty neat, especially due to the fact wlvncc is still in development and it might require unstable versions of libs that conflict with the already installed stable libs.

Note that both `commit` and `branch` are set in order to catch new commits on rebuilds so build will fail when there are new commits.  
Prerequisites are `flatpak-builder` and the `org.freedesktop.Sdk//20.08` runtime installed.  
```
$ flatpak install flathub org.freedesktop.Sdk//20.08
```
To build and install just run something like:
```
$ export wrkdir=$PWD/wrkdir # or anywhere else
$ flatpak-builder \
  --install \
  --ccache \
  --user \
  --force-clean \
  --state-dir=${wrkdir}/flatpak-builder \
  --repo=${wrkdir}/flatpak-repo  \
  ${wrkdir}/flatpak-target \
  com.github.any1.wlvncc.yaml
```
And then run with `$ com.github.any1.wlvncc ...` or something like
```
$ flatpak run --user com.github.any1.wlvncc ...
$ flatpak run --user --devel --command=sh com.github.any1.wlvncc
```

**Comment by any1** [comment]
Thanks! Luckily GitHub added a new "Discussions" feature where we can move this.

**Comment by tinywrkb** [comment]
I posted a packaged Flatpak bundle if anyone wants to test this and is too lazy to build.  
Just download the file and install the package with your package manager (Gnome Software? KDE Discover?) or from the command-line like this:

```
$ flatpak --user install com.github.any1.wlvncc.r43.1ca82ce.flatpak
```

https://github.com/tinywrkb/flatpaks/releases/tag/wlvncc-r43.1ca82ce

**Comment by any1** [comment]
Perhaps I ought to mention that if wlvncc is compiled with libvncserver and aml as subprojects, they will be statically linked into the binary.

**Comment by tinywrkb** [comment]
Oh, that's nice, I didn't notice this.
